File: SereneEmployees/SereneEmployees.Web/Modules/ServerTypes/Personnel/EmployeeForm.ts
import { StringEditor, DateEditor, DecimalEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface EmployeeForm {
    FullName: StringEditor;
    HireDate: DateEditor;
    BirthDate: DateEditor;
    Salary: DecimalEditor;
    Image: ImageUploadEditor;
}

export class EmployeeForm extends PrefixedContext {
    static readonly formKey = 'Personnel.Employee';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeForm.init)  {
            EmployeeForm.init = true;

            var w0 = StringEditor;
            var w1 = DateEditor;
            var w2 = DecimalEditor;
            var w3 = ImageUploadEditor;

            initFormType(EmployeeForm, [
                'FullName', w0,
                'HireDate', w1,
                'BirthDate', w1,
                'Salary', w2,
                'Image', w3
            ]);
        }
    }
}

File: SereneEmployees/SereneEmployees.Web/Modules/Personnel/Employee/EmployeeForm.cs
using Serenity.ComponentModel;
using System;

namespace SereneEmployees.Personnel.Forms;

[FormScript("Personnel.Employee")]
[BasedOnRow(typeof(EmployeeRow), CheckNames = true)]
public class EmployeeForm
{
    public string FullName { get; set; }
    public DateTime HireDate { get; set; }
    public DateTime BirthDate { get; set; }
    public decimal Salary { get; set; }
    public string Image { get; set; }
}